### PR TITLE
Rename DeploymentName

### PR DESCRIPTION
**Pull request by carlydf**
## What was changed
Rename DeploymentName to DeploymentSeriesName to match api-go/versioning-3 branch

## Why?
So that my versioning-3 code will compile in temporal server

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

**Review by ShahabT — APPROVED** [review]

**Comment by antlai-temporal** [comment]
@carlydf and @ShahabT before merging into the sdk-go versioning-3 branch can you give me a heads up first, or at least wait for someone from the SDK team to approve the changes. Thanks.

**Pull request merged**
